### ISSUE TITLE
Access validation data from callback

### ISSUE THREAD

**Issue by cklat** [matterport/Mask_RCNN#547]
Hi,

I want to train my own dataset and need to write my own metric for this. I'm planning to this with a custom callback (inherited from keras.callbacks.Callback). However, I couldn't figure out how to access the validation (and train) data that is used in the current epoch when the model is training. Could you give me some hints on where to look in the code? 

Also, if there is an easier way to calculate and print own metrics, I'm absolutely open for any ideas. Thank you!
